src/extractor/extraction_containers.cpp

```diff
@@ -453,11 +453,13 @@ void ExtractionContainers::PrepareEdges(lua_State *segment_state)
                 all_edges_list[i].result.weight < min_forward_weight)
             {
                 min_forward_idx = i;
+                min_forward_weight = all_edges_list[i].result.weight;
```

Comment:
Yes, without the change min_forward_weight is always std::numeric_limits<int>::max()
and "all_edges_list[i].result.weight < min_forward_weight" is always true